rfcs/build-system/04-build-shield.md

```diff
@@ -0,0 +1,107 @@
+| ----------------------------------------------- | ---------------------------------------------------------------------------- |
+| Release build failed                            | 🔥 Investigate ASAP (fix issue, or retry if intermittent)                    |
+| CI _consistently_ broken                        | 🔥 Fix ASAP                                                                  |
+| PR builds _consistently_ broken (CI okay)       | 🔥 Fix ASAP                                                                  |
+| Local builds _consistently_ broken (PR/CI okay) | 🔥 Fix ASAP                                                                  |
+| Published package broken                        | 🔥 Fix ASAP (usually due to dep issue or missing file)                       |
+| CI _intermittently_ broken                      | ⚠️ Try to investigate/fix, or file an issue if it will take significant time |
+| v7/8 website release failed (uncommon)          | ⚠️ Re-run failed stage, or see internal wiki                                 |
+
+#### Documenting issues and troubleshooting procedures
+
+The second responsibility is **documenting** failures and troubleshooting procedures, with the goals of:
+
+- Noticing failure patterns which we need to schedule work to address
+- Establishing a "runbook" for troubleshooting build issues
+  - The goal of this is to de-mystify build, and maybe even (eventually) be able to expand the build shield rotation to more team members, so that everyone has exposure to the area
+
+Where/how to document:
+
+- Troubleshooting guides should go in the internal wiki
+- Comment on automated failure posts with the cause of the failure and the resolution
+  - We may also want to make a form or spreadsheet or something to more easily gather metrics on failures
+
+### Non-responsibilities (initially)
+
+At least initially, build shield is NOT responsible for general build improvements/maintenance, such as:
+
+- Upgrading deps
+- Addressing non-critical dependabot/governance alerts
+- Perf improvements
+- Non-urgent build-related requests or proposals that come through github issues
+
+(If time allows and you'd like to work on any of these things, that's great, but it's not the primary expectation.)
+
+We may in the future make build shield a larger responsibility (with larger time allocation) to address these types of things in a more systematic manner. But to make this a feasible, incremental improvement/experiment, we're keeping the scope small initially.
+
+## Pros and Cons
+
+### Pros
+
+- Less randomization of specific people
+- More people learn about build troubleshooting
+
+### Cons
+
+- It's another process to manage
+
+## Discarded Solutions
+
+The regular shield person has more than enough to do already, so it probably doesn't make sense to add this to their responsibilities.
+
+## Open Issues
+
+- Specific list of people on the rotation
```

Comment:
Here's the list I'd suggest. I nominate @TristanWatanabe to go first on the Redmond side (I'll help) and Prague people can discuss among themselves.

- Redmond:
  - @ecraig12345 
  - @TristanWatanabe 
  - if we try to have 1 per time zone, ideally we should add at least one more person (any suggestions @JustSlone?)
- Prague:
  - @ling1726 
  - @Hotell 
  - @varholak-peter 
  - @andrefcdias

Reply:
Let's start with @ecraig12345 and @TristanWatanabe on Redmond shield, we can add more folks over time.